Lib/test/test_inspect/test_inspect.py

```diff
@@ -1872,6 +1895,12 @@ def static_method(): pass
             def a_property(self): pass
         class Slotermeyer:
             __slots__ = 'a_slot',
+        member_descriptor = Slotermeyer.__dict__['a_slot']
+        assert (inspect.ismemberdescriptor(member_descriptor)
+                or not hasattr(types, 'MemberDescriptorType'))
+        getset_descriptor = types.FrameType.f_locals
+        assert (inspect.isgetsetdescriptor(getset_descriptor)
+                or not hasattr(types, 'GetSetDescriptorType'))
```

Comment:
Use real if-else-guards with assertTrue or assertFalse depending on the result. Ideally, another test that is `@cpython_only` would be preferrable so that other implementations that run the test suite don't fail in this case.